Unable to build GC benchmarks

I'm unable to build the GC benchmarks using [these instructions](https://github.com/dotnet/performance/tree/main/src/benchmarks/gc). There appears to be a parsing error when I run the `python3 . setup` command:

```
$ python3 . setup
Traceback (most recent call last):
File "/usr/lib/python3.8/runpy.py", line 194, in _run_module_as_main
return _run_code(code, main_globals, None,
File "/usr/lib/python3.8/runpy.py", line 87, in _run_code
exec(code, run_globals)
File "./__main__.py", line 9, in <module>
run_command(ALL_COMMANDS)
File "./src/commonlib/command.py", line 87, in run_command
run_command_worker(commands, cmd_and_args)
File "./src/commonlib/command.py", line 233, in run_command_worker
cast(Callable[[], None], command.fn)()
File "./src/commonlib/setup.py", line 26, in setup
write_host_info()
File "./src/commonlib/host_info.py", line 276, in write_host_info
write_yaml_file(HOST_INFO_PATH, _get_host_info(built), overwrite=True)
File "./src/commonlib/host_info.py", line 114, in _get_host_info
return {OS.posix: lambda _: _get_host_info_posix(), OS.windows: _get_host_info_windows}[
File "./src/commonlib/host_info.py", line 114, in <lambda>
return {OS.posix: lambda _: _get_host_info_posix(), OS.windows: _get_host_info_windows}[
File "./src/commonlib/host_info.py", line 152, in _get_host_info_posix
l1d_cache_kb = get_opt_kb("L1d cache")
File "./src/commonlib/host_info.py", line 139, in get_opt_kb
return map_option(opt, lambda s: int(remove_str_end(s, "K")))
File "./src/commonlib/option.py", line 25, in map_option
return None if t is None else cb(t)
File "./src/commonlib/host_info.py", line 139, in <lambda>
return map_option(opt, lambda s: int(remove_str_end(s, "K")))
File "./src/commonlib/util.py", line 43, in remove_str_end
assert s.endswith(end), f"Expected {s} to end with {end}"
AssertionError: Expected 4 MiB to end with K 
```

Looks like it's failing when trying to parse `lscpu` output. Relevant section of `lscpu` output:

```
L1d cache:                       4 MiB
L1i cache:                       4 MiB
L2 cache:                        64 MiB
L3 cache:                        512 MiB
```

--- a/src/benchmarks/gc/src/commonlib/host_info.py
+++ b/src/benchmarks/gc/src/commonlib/host_info.py
@@ -27,10 +27,12 @@ from .util import (
     ensure_dir,
     ExecArgs,
     exec_and_get_output,
+    gb_to_kb,
     get_hostname,
     get_os,
     kb_to_bytes,
     kb_to_mb,
+    mb_to_kb,
     mhz_to_ghz,
     OS,
     remove_str_end,
@@ -136,7 +138,31 @@ def _get_host_info_posix() -> HostInfo:
         opt = get_opt(name)
         if opt is not None and _UNKNOWN_MSG in opt.lower():
             return None
-        return map_option(opt, lambda s: int(remove_str_end(s, "K")))
+
+        # Retrieve the size of the given memory option string, and return it
+        # in KB. Nowadays, it is not uncommon for machines to have MB, or in
+        # more powerful cases, GB of cache, so we need to do the conversion
+        # as necessary since the infra expects to receive KB.
+        #
+        # First we extract the size units name to determine how to convert them
+        # if necessary. Since we don't know about every single machine's rules
+        # for capitalization, we'll convert to lowercase just to be safe.
+
+        assert opt is not None
+        size_units = opt.rsplit(" ", 1)[-1].lower()
+        converted_kb = 0.0
+
+        if size_units in ["k", "kib"]:
+            converted_kb = int(remove_str_end(opt.lower(), size_units))
+            return map_option(converted_kb, lambda n: n)
+        elif size_units in ["m", "mib"]:
+            converted_kb = mb_to_kb(float(remove_str_end(opt.lower(), size_units)))
+            return map_option(converted_kb, lambda n: int(n))  # pylint: disable=W0108
+        elif size_units in ["g", "gib"]:
+            converted_kb = gb_to_kb(float(remove_str_end(opt.lower(), size_units)))
+            return map_option(converted_kb, lambda n: int(n))  # pylint: disable=W0108
+        else:
+            raise Exception(f"Unrecognized size units '{size_units}'")
 
     # Note: "CPU MHz" is the *current* cpu rate which varies. Going with max here.
     # TODO: Max is probably wrong, we want a typical value.

--- a/src/benchmarks/gc/src/commonlib/util.py
+++ b/src/benchmarks/gc/src/commonlib/util.py
@@ -608,8 +608,16 @@ def kb_to_mb(kb: float) -> float:
     return bytes_to_mb(kb_to_bytes(kb))
 
 
+def mb_to_kb(mb: float) -> float:
+    return bytes_to_kb(mb_to_bytes(mb))
+
+
 def mb_to_gb(mb: float) -> float:
     return bytes_to_gb(mb_to_bytes(mb))
+
+
+def gb_to_kb(gb: float) -> float:
+    return bytes_to_kb(gb_to_bytes(gb))
 
 
 def gb_to_mb(gb: float) -> float:

